internal/jimm/model.go

```diff
@@ -748,16 +749,103 @@ func (j *JIMM) ModelInfo(ctx context.Context, user *openfga.User, mt names.Model
 	return j.mergeModelInfo(ctx, user, mi, m)
 }
 
+type modelSummariesSafeMap struct {
+	mu             sync.Mutex
+	modelSummaries map[string]jujuparams.ModelSummaryResult
+}
+
+func (m *modelSummariesSafeMap) addModelSummary(summary jujuparams.ModelSummaryResult) {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+	if m.modelSummaries == nil {
+		m.modelSummaries = make(map[string]jujuparams.ModelSummaryResult)
+	}
+	m.modelSummaries[summary.Result.UUID] = summary
+}
+
+// ModelSummaries returns the list of modelsummary the user has access to.
+// It queries the controllers and then merge the info from the JIMM db.
+func (j *JIMM) ModelSummaries(ctx context.Context, user *openfga.User, maskingControllerUUID string) (jujuparams.ModelSummaryResults, error) {
+	const op = errors.Op("jimm.ModelSummaries")
+
+	modelSummariesSafeMap := modelSummariesSafeMap{}
+	modelSummaryResults := []jujuparams.ModelSummaryResult{}
+
+	var models []struct {
+		model      *dbmodel.Model
+		userAccess jujuparams.UserAccessPermission
+	}
+	// we collect models belonging to the user.
+	err := j.ForEachUserModel(ctx, user, func(m *dbmodel.Model, uap jujuparams.UserAccessPermission) error {
+		models = append(models, struct {
+			model      *dbmodel.Model
+			userAccess jujuparams.UserAccessPermission
+		}{model: m, userAccess: uap})
+		return nil
```

Comment:
you could assemble the list of unique controllers in this loop.. no need to have a separate for loop that ranges over models

Reply:
no?

Reply:
i did it, just forgot to resolve the comment. Sorry